authority/provisioner/jwk.go

```diff
@@ -0,0 +1,127 @@
+package provisioner
+
+import (
+	"crypto/x509"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/crypto/x509util"
+	"github.com/smallstep/cli/jose"
+)
+
+// jwtPayload extends jwt.Claims with step attributes.
+type jwtPayload struct {
+	jose.Claims
+	SANs []string `json:"sans,omitempty"`
+}
+
+// JWK is the default provisioner, an entity that can sign tokens necessary for
+// signature requests.
+type JWK struct {
```

Comment:
You've put the audiences is into the provisioners. Wondering if that's going to cause issues with the revocation audiences being different than the sign audiences. Will just be something we'll need to figure out when integrating my stuff.

Reply:
I already noted this in your PR, we have to integrate it into my branch
https://github.com/smallstep/certificates/pull/50#discussion_r264937336